src/material/button/_button-theme.scss

```diff
@@ -33,15 +33,27 @@ $_mat-button-ripple-opacity: 0.1;
   $warn: map-get($theme, warn);
 
   &.mat-primary .mat-ripple-element {
-    background-color: mat-color($primary, $hue, $opacity);
+    $color: mat-color($primary, $hue, $opacity);
+    background-color: $color;
+    @if (type_of($color) != color) {
+      opacity: $opacity;
+    }
```

Comment:
Instead of repeating this block three times, could you introduce a mixin in this file that all three locations use along with a comment that explains the context?

Reply:
@jelbourn Thanks for pointing that out. I have made the update.